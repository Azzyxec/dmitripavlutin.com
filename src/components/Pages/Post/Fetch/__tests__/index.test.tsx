import * as React from 'react';
import { shallow } from 'enzyme';

import PostFetch from '../index';

const siteInfo = {
  title: 'Dmitri Pavlutin blog',
  description: 'Posts by Dmitri Pavlutin about software development',
  url: 'https://dmitripavlutin.com',
  repositoryUrl: 'https://github.com/panzerdp/dmitripavlutin.com',
};

const authorProfilePicture = {
  width: 100,
  height: 100,
  base64: 'base64-encoded-string',
  src: '/image.png',
  srcSet: 'some srcset values',
};

const authorInfo = {
  name: 'Dmitri Pavlutin',
  description: 'Dmitri Pavlutin is a software developer specialized in Frontend technologies',
  email: 'mail@mail.com',
  profiles: {
    stackoverflow: 'https://stackoverflow.com/users/1894471/dmitri-pavlutin',
    twitter: 'https://twitter.com/panzerdp',
    linkedin: 'https://www.linkedin.com/in/dmitri-pavlutin/',
    github: 'https://github.com/panzerdp',
    facebook: 'https://www.facebook.com/dmitri.pavlutin',
  },
  nicknames: {
    twitter: 'panzerdp',
  },
};

const props = {
  data: {
    site: {
      siteMetadata: {
        siteInfo,
        authorInfo,
      },
    },
    authorProfilePicture: {
      childImageSharp: {
        resize: authorProfilePicture,
      },
    },
    recommendedPosts: {
      edges: [
        {
          node: {
            frontmatter: {
              title: 'Useful techniques to facilitate React server-side rendering',
              description: 'Useful techniques to facilitate React server-side rendering.',
              published: '2018-03-17',
              slug: 'useful-techniques-react-server-side-rendering',
              tags: ['react', 'server side rendering'],
              thumbnail: {
                childImageSharp: {
                  fluid: {
                    src: 'source',
                    base64: 'base64',
                    aspectRatio: 2,
                    srcSet: 'src-set',
                    sizes: 'some sizes',
                    srcWebp: 'src-webp',
                    srcSetWebp: 'src-set-webp',
                  },
                },
              },
            },
          },
        },
      ],
    },
    popularPosts: {
      edges: [
        {
          node: {
            frontmatter: {
              title: 'Useful techniques to facilitate React server-side rendering',
              description: 'Useful techniques to facilitate React server-side rendering.',
              published: '2018-03-17',
              slug: 'useful-techniques-react-server-side-rendering',
              tags: ['react', 'server side rendering'],
            },
          },
        },
      ],
    },
    markdownRemark: {
      id: 'post-id',
      html: '<div>JavaScript is a programming language</div>',
      fileAbsolutePath: '/home/user/my-post.md',
      frontmatter: {
        title: 'Useful techniques to facilitate React server-side rendering',
        description: 'Useful techniques to facilitate React server-side rendering.',
        published: '2018-03-17',
        modified: '2019-01-01',
        slug: 'useful-techniques-react-server-side-rendering',
        tags: ['react', 'server side rendering'],
        recommended: ['javascript-language'],
        thumbnail: {
          childImageSharp: {
            fluid: {
              src: 'source',
              base64: 'base64',
              aspectRatio: 2,
              srcSet: 'src-set',
              sizes: 'some sizes',
              srcWebp: 'src-webp',
              srcSetWebp: 'src-set-webp',
            },
          },
        },
      },
    },
  },
};

describe('<PostFetch />', function() {
  it('should render post template', function() {
    const wrapper = shallow(<PostFetch {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
