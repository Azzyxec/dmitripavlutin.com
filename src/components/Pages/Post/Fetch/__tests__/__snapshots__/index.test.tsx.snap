// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostFetch /> should render post template 1`] = `
<PostTemplate
  authorInfo={
    Object {
      "description": "Dmitri Pavlutin is a software developer specialized in Frontend technologies",
      "name": "Dmitri Pavlutin",
      "nicknames": Object {
        "twitter": "panzerdp",
      },
      "profiles": Object {
        "facebook": "https://www.facebook.com/dmitri.pavlutin",
        "github": "https://github.com/panzerdp",
        "linkedin": "https://www.linkedin.com/in/dmitri-pavlutin/",
        "stackoverflow": "https://stackoverflow.com/users/1894471/dmitri-pavlutin",
        "twitter": "https://twitter.com/panzerdp",
      },
    }
  }
  authorProfilePictureSrc="/image.png"
  post={
    Object {
      "description": "Useful techniques to facilitate React server-side rendering.",
      "html": "<div>JavaScript is a programming language</div>",
      "modified": "2019-01-01",
      "published": "2018-03-17",
      "recommended": Array [
        "javascript-language",
      ],
      "slug": "useful-techniques-react-server-side-rendering",
      "tags": Array [
        "react",
        "server side rendering",
      ],
      "thumbnail": Object {
        "aspectRatio": 2,
        "base64": "base64",
        "sizes": "some sizes",
        "src": "source",
        "srcSet": "src-set",
      },
      "title": "Useful techniques to facilitate React server-side rendering",
    }
  }
  postRepositoryFileUrl="https://github.com/panzerdp/dmitripavlutin.com/tree/master//home/user/my-post.md"
  recommendedPosts={
    Array [
      Object {
        "description": "Useful techniques to facilitate React server-side rendering.",
        "published": "2018-03-17",
        "slug": "useful-techniques-react-server-side-rendering",
        "tags": Array [
          "react",
          "server side rendering",
        ],
        "thumbnail": Object {
          "aspectRatio": 2,
          "base64": "base64",
          "sizes": "some sizes",
          "src": "source",
          "srcSet": "src-set",
        },
        "title": "Useful techniques to facilitate React server-side rendering",
      },
    ]
  }
  siteInfo={
    Object {
      "description": "Posts by Dmitri Pavlutin about software development",
      "repositoryUrl": "https://github.com/panzerdp/dmitripavlutin.com",
      "title": "Dmitri Pavlutin blog",
      "url": "https://dmitripavlutin.com",
    }
  }
/>
`;
